modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/history/upnext/UpNextHistoryDetailsPage.kt

```diff
@@ -0,0 +1,245 @@
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import au.com.shiftyjelly.pocketcasts.compose.AppThemeWithBackground
+import au.com.shiftyjelly.pocketcasts.compose.Devices
+import au.com.shiftyjelly.pocketcasts.compose.bars.ThemedTopAppBar
+import au.com.shiftyjelly.pocketcasts.compose.components.EpisodeImage
+import au.com.shiftyjelly.pocketcasts.compose.components.TextH40
+import au.com.shiftyjelly.pocketcasts.compose.components.TextH50
+import au.com.shiftyjelly.pocketcasts.compose.components.TextH70
+import au.com.shiftyjelly.pocketcasts.compose.loading.LoadingView
+import au.com.shiftyjelly.pocketcasts.compose.preview.ThemePreviewParameterProvider
+import au.com.shiftyjelly.pocketcasts.localization.helper.RelativeDateFormatter
+import au.com.shiftyjelly.pocketcasts.localization.helper.TimeHelper
+import au.com.shiftyjelly.pocketcasts.models.entity.BaseEpisode
+import au.com.shiftyjelly.pocketcasts.models.entity.PodcastEpisode
+import au.com.shiftyjelly.pocketcasts.settings.history.upnext.UpNextHistoryDetailsViewModel.UiState
+import au.com.shiftyjelly.pocketcasts.ui.theme.Theme
+import java.text.DateFormat
+import java.util.Date
+import au.com.shiftyjelly.pocketcasts.localization.R as LR
+
+@Composable
+fun UpNextHistoryDetailsPage(
+    viewModel: UpNextHistoryDetailsViewModel = hiltViewModel(),
+    date: Long,
+    onRestoreClick: (restoreUpNext: () -> Unit) -> Unit,
+    onBackClick: () -> Unit,
+    bottomInset: Dp,
+) {
+    val state by viewModel.state.collectAsStateWithLifecycle()
+    UpNextHistoryDetailsView(
+        date = date,
+        state = state,
+        onRestoreClick = { onRestoreClick { viewModel.restoreUpNext() } },
+        onBackClick = onBackClick,
+        bottomInset = bottomInset,
+    )
+}
+
+@Composable
+private fun UpNextHistoryDetailsView(
+    date: Long,
+    state: UiState,
+    onRestoreClick: () -> Unit,
+    onBackClick: () -> Unit,
+    bottomInset: Dp,
+) {
+    Column {
+        ThemedTopAppBar(
+            title = formatDate(date),
+            bottomShadow = true,
```

Comment:
```suggestion
```

I think it looks more modern to not have the drop shadow. Feel free to ignore if you disagree.